data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/plugins/sink/LatencyMetrics.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink;
+
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import io.micrometer.core.instrument.DistributionSummary;
+import org.opensearch.dataprepper.model.event.EventHandle;
+
+import java.time.Duration;
+import java.time.Instant;
+
+public class LatencyMetrics {
+    public static final String INTERNAL_LATENCY = "internalLatency";
+    public static final String EXTERNAL_LATENCY = "externalLatency";
+    private final DistributionSummary internalLatencySummary;
+    private final DistributionSummary externalLatencySummary;
+
+    public LatencyMetrics(PluginMetrics pluginMetrics) {
+        internalLatencySummary = pluginMetrics.summary(INTERNAL_LATENCY);
+        externalLatencySummary = pluginMetrics.summary(EXTERNAL_LATENCY);
+    }
+    public void update(final EventHandle eventHandle) {
```

Comment:
I tend to think that this should take the `result` passed into the `EventHandle` and make the decision as to whether or not to record here.

```
public void update(final EventHandle eventHandle, final boolean result) {
  if(result == false)
    return;
  ...
}
```